core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -135,6 +135,11 @@ private[spark] class BlockManager(
   // Whether to compress shuffle output temporarily spilled to disk
   private val compressShuffleSpill = conf.getBoolean("spark.shuffle.spill.compress", true)
 
+  // Max number of failures before this block manager refreshes the block locations from the driver
+  private val maxFailuresBeforeLocationRefresh =
+    conf.getInt("spark.block.failures.beforeLocationRefresh", Int.MaxValue)
```

Comment:
I would just make this a constant instead of introducing another config:
```
private[spark] object BlockManager {
  /**
   * Max number of failures when fetching a block remotely before refreshing the block's
   * locations from the driver. This is useful for cases where many executors fail at once,
   * e.g. dynamic allocation. (SPARK-13328)
   */
  private val REFRESH_LOCATION_FETCH_FAILURE_THRESHOLD = 10
}
```